### DISCUSSION TITLE
Hydration completed but contains mismatches.

### DISCUSSION THREAD

**Discussion by Beny2305** [vuepress-theme-hope/vuepress-theme-hope#3295] [Q&A] [answered]
client报错 Hydration completed but contains mismatches.   SSR生成的html中包含了<aside><main><footer>三个标签的及内容，客户端运行时重复插入以上三个标签及内容，有人遇到过吗？
替换了Navbar组件，取消替换就能正常，但不知道问题在哪里，什么情况触发的mismatches

**Comment by Mister-Hope** [comment, marked as answer]
这是你自己的代码问题，贴这几个字是指望别人给你算一卦么？

**Comment by Beny2305** [comment]
是的，我也不知道哪里的问题，用标签重写了navtab就组件就有问题，应该是SSR生成的代码和客户端展现的不一致，用<ClientOnly>标签把navtab包起来就没了

Mr.Hope ***@***.***> 于2023年9月5日周二 13:28写道：

**Reply by Mister-Hope** [reply]
是的用clientonly 的确是一个很有效的方法，但是你正道还是自己排查，哪里会在浏览器和Node端导致渲染不一致？

**Comment by Beny2305** [comment]
我并不是专做前端的，赶工就没时间去细查了。
查阅vuepress
SSR相关部分的文档，记得大概意思是为了让node渲染的内容和浏览器看到的一致，会进行某种验证，目的是为了服务端和客户端之间的数据传输不出问题，如果验证不一致就会出现这个报错。



Mr.Hope ***@***.***> 于2023年9月7日周四 20:54写道：
